packages/next/client/index.js

```diff
@@ -536,5 +550,13 @@ async function doRender({ App, Component, props, err }) {
     appElement
   )
 
+  await renderPromise
   emitter.emit('after-reactdom-render', { Component, ErrorComponent, appProps })
 }
+
+function Root({ callback, children }) {
+  React.useLayoutEffect(() => {
```

Comment:
Is there any reason to prefer `useLayoutEffect` over `useEffect` here? Could you elaborate on why with a comment (or switch to `useEffect`)?